src/pages/admin/index.tsx

```diff
@@ -0,0 +1,50 @@
+import * as React from 'react';
+import Router from 'next/router';
+import { useSession } from 'next-auth/react';
+import { QRGenerator } from '../../components/App';
+import { isAdmin } from '../../services/admin';
+
+const Admin = () => {
+  const { data: session, status } = useSession();
+  const [validAdmin, setValidAdmin] = React.useState(false);
+
+  React.useEffect(() => {
+    if (status === 'unauthenticated') {
+      Router.push('/auth/login');
+    }
+  }, [status]);
+
+  const email = session?.user?.email;
+  React.useEffect(() => {
+    const checkAdmin = async () => {
+      setValidAdmin(await isAdmin(email));
+    };
+    checkAdmin();
+  }, [email]);
+
+  if (status === 'loading') {
+    return <div>Loading...</div>;
+  }
+
+  if (status === 'authenticated') {
+    return validAdmin ? (
+      <>
+        <h1> ADMIN PAGE 🤓 </h1>
+        {/** TODO: Create Header for admin page  */}
+        <QRGenerator />
+      </>
+    ) : (
+      <h1> You are not allowed here </h1>
+    ); // This will need adjustment since it's just a prototype
+  }
+
+  return (
+    <>
+      {/** TODO: Create Header for admin page  */}
```

Comment:
Do we need this, the code seems to redirect unauthenticated users back to the login page anyway.

It's worth having a version of the 403 component that handles what this displays

Reply:
Yeah we should just 403 if they're not authorized